File: 백준/Silver/20920. 영단어 암기는 괴로워/영단어 암기는 괴로워.py
import sys
input = sys.stdin.readline

word_dict = {}

n, m = map(int, input().split())

for _ in range(n):
    word = input().rstrip()
    if len(word) < m:
        continue
    if word not in word_dict:
        word_dict[word] = 1
    else:
        word_dict[word] += 1

word_lst = sorted(word_dict.items(), key = lambda x: (-x[1], -len(x[0]), x[0]))
for word, count in word_lst: print(word)

File: 백준/Silver/20920. 영단어 암기는 괴로워/영단어 암기는 괴로워.swift
import Foundation

let nm = readLine()!.split(separator: " ").map { Int($0)! }
let n = nm[0]
let m = nm[1]

var wordDict: [String: Int] = [:]

for _ in 0..<n {
    let word = readLine()!
    if word.count < m { continue}
    wordDict[word, default: 0] += 1
}

let sortedWords = wordDict.sorted { a, b in
    if a.value != b.value {
        return a.value > b.value
    }
    
    if a.key.count != b.key.count {
        return a.key.count > b.key.count
    }
    
    return a.key < b.key
}

var output = ""
for (word, _) in sortedWords {
    output += word + "\n"
}

print(output)
